### PR TITLE
Upgrade to Postgres 16

### PR DESCRIPTION
**Pull request by jamesbiggs**
Ticket URL: [EDEV-127]

## About these changes

Upgrade our local environment and Platform.sh to Postgres 16

## How to check these changes

Add `POSTGRES_HOST_AUTH_METHOD=trust` to your `.env` file.

Delete your Docker containers, images, and volumes.

`fab build && fab start && fab pull-staging`

Also check the `postgres-upgrade` environment on Platform.sh

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [x] Ensured that PR includes only commits relevant to the ticket
- [x] Waited for all CI jobs to pass before requesting a review
- [x] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

[EDEV-127]: https://national-archives.atlassian.net/browse/EDEV-127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 
   db:
-    image: postgres:12.0
+    image: postgres:16.0
     ports:
       - 5432:5432
     volumes:
       - ./database_dumps:/app/database_dumps
+    environment:
+      - POSTGRES_PASSWORD=postgres
```
Comment (TNA-Allan) [inline docker-compose.yml:10]: To make `fab pull-staging-data` work:

```suggestion
    env_file:
          - .env
    environment:
      - POSTGRES_PASSWORD=postgres
```

Add `POSTGRES_HOST_AUTH_METHOD=trust` to `.env`, but not on the hosted environment

File: docker-compose.yml [hunk 2]
```diff
@@ -39,6 +41,8 @@ services:
       - DATABASE_HOST=db
       - DATABASE_NAME=postgres
       - DATABASE_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - DATABASE_PASSWORD=postgres
```
Comment (TNA-Allan) [inline docker-compose.yml:45]: To make `fab pull-staging-data` work

```suggestion
```

File: docker-compose.yml [hunk 3]
```diff
@@ -81,6 +85,8 @@ services:
...
```
Comment (TNA-Allan) [inline docker-compose.yml:89]: To make `fab pull-staging-data` work
```suggestion
```

**Review by TNA-Allan — COMMENTED** [review]
Hi @jamesbiggs 
Regarding issue with `fab pull-staging-data` causing to prompt for password twice, it is in the   `dropdb, createdb` commands.
I've suggested some changes in the `docker-compose.yml` and `.env` for local only which bypasses the prompts - it worked for me.

File: .env.example [hunk 4]
```diff
@@ -2,6 +2,7 @@ DATABASE_ENGINE=django.db.backends.postgresql
 DATABASE_NAME=postgres
 DATABASE_USER=postgres
 DATABASE_HOST=db
+POSTGRES_HOST_AUTH_METHOD=trust
```
Comment (ahosgood) [inline .env.example:5]: Can we move this into the `docker-compose.yml`? It's nothing sensitive and needs to be in there for the dev env to work.

File: docker-compose.yml [hunk 5]
```diff
@@ -39,6 +43,8 @@ services:
       - DATABASE_HOST=db
       - DATABASE_NAME=postgres
       - DATABASE_USER=postgres
+      - POSTGRES_PASSWORD=postgres
```
Comment (ahosgood) [inline docker-compose.yml:46]: Isn't this a duplicate?

Comment (ahosgood) [reply docker-compose.yml:46]: As with the line below...

File: .env.example (hunk 4 above)
Comment (jamesbiggs) [reply .env.example:5]: We can but won't we have to move it again when we start using Docker on AWS?

**Review by ahosgood — APPROVED** [review]
⚠️⚠️⚠️ This looks fine but I don't think we should merge it in until the dev fix has gone up to live: https://github.com/nationalarchives/ds-wagtail/compare/main...develop ⚠️⚠️⚠️

**Pull request merged**